src/basic/sysctl-util.c

```diff
@@ -56,12 +76,18 @@ int sysctl_write(const char *property, const char *value) {
         if (!path_is_normalized(p))
                 return -EINVAL;
 
+        ret = sysctl_f_ofd_lk(p, lock);
+        if (ret < 0) {
+                log_warning("Skipping sysctl write to '%s', it's handled elsewhere.", p);
+                return -EBUSY;
+        }
+
         log_debug("Setting '%s' to '%s'", p, value);
 
         return write_string_file(p, value, WRITE_STRING_FILE_VERIFY_ON_FAILURE | WRITE_STRING_FILE_DISABLE_BUFFER | WRITE_STRING_FILE_SUPPRESS_REDUNDANT_VIRTUAL);
```

Comment:
hmm, so if implemented this way we'll first take a lock on the file, then give it up agan, and then reopen the file, and write to it. That is a bit racy. Not really a race that would matter, but it still has a bit of a smell, I think.

I'd prefer a different approach:

add a new flag WRITE_STRING_FILE_UNPOSIX_WRITE_LOCK to the write_string_file() helper. Ill this would do is lock the file right after opening the file, and before writing to it. That removes the race, and makes the logic generic.

This would be a few lines added to write_string_file_ts_at() in src/basic/fileio.c